src/ESports.ts

```diff
@@ -0,0 +1,152 @@
+        if (args.length !== 1) return;
+
+        // day / month
+        const formatCheck = new RegExp("\\d\\d/\\d\\d");
+        if (!formatCheck.test(args[0])) return;
+
+        const data = args[0].split("/");
+        const date = `${data[0]} ${data[1]} ${new Date().getFullYear()}`;
+
+        this.sendPrintout(message.channel as Discord.TextChannel, this.schedule[date], date);
+    }
+
+    private postInfo() {
+        const esports = this.bot.guilds.get(this.settings.server)!.channels.find("name", "esports-spoilers");
+        if (!esports) {
+            console.error(`Esports: Unable to find channel #esports-spoilers`);
+            return;
+        }
+
+        const now = new Date();
+        const date = `${now.getDate()} ${("0" + (now.getMonth() + 1)).slice(-2)} ${now.getFullYear()}`;
+
+        // filter old games
+        const prints: { [league: string]: ESportsLeagueSchedule[] } = {};
+        Object.keys(this.schedule[date]).forEach(league => {
+            prints[league] = this.schedule[date][league].filter(e => e.time.indexOf("ago") === -1);
+        });
+
+        this.sendPrintout(esports as Discord.TextChannel, prints, date);
+        setTimeout(this.postInfo, 3600000);
+    }
+
+    private sendPrintout(channel: Discord.TextChannel, data: { [league: string]: ESportsLeagueSchedule[] }, date: string) {
+        let output = `Games being played ${date.split(" ").join("/")}:\n\`\`\``;
+        const padLeague = Object.keys(data).reduce((a, b) => a.length > b.length ? a : b).length;
+        Object.keys(data).forEach(league => {
+            const games = data[league];
+            for (const game of games) {
+                output += `[${league.padEnd(padLeague)}] ${game.teamA.padEnd(3)} vs ${game.teamB.padEnd(3)} -- ${game.time}\n`;
+            }
+        });
+        output += "```";
+
+        channel.send(output);
+    }
+
+    // this can also check for older games by using resultsHtml instead of fixures
+    private async loadData() {
+        // pull data
+        const data = await fetch("https://eu.lolesports.com/en/api/widget/schedule?timezone=Europe%2FOslo&leagues=26&leagues=3&leagues=2&leagues=6&leagues=7&leagues=5&leagues=4&leagues=9&leagues=10&leagues=1&leagues=43&slug=all");
+        const html = (await data.json() as ESportsAPIReturnData).fixturesHtml;
+        const schedule: { [date: string]: { [league: string]: ESportsLeagueSchedule[] } } = {};
+
+        // for each date
+        const asHtml = CheerioAPI.load(html);
+        const parents = asHtml.root().find(".schedule__row-group");
+        parents.each((indexed, elem) => {
+            // the current date
+            const elemRoot = CheerioAPI.load(elem).root();
+            const date = elemRoot.find(".schedule__row--date").find("h2").text().split(" ");
+
+            // hack to get month number from text
+            const month = ("0" + ("JanFebMarAprMayJunJulAugSepOctNovDec".indexOf(date[2].substr(0, 3)) / 3 + 1)).slice(-2);
```

Comment:
what the fuck

Reply:
If you have a better way of turning a string date into a number ill take it 🤷‍♂️

Reply:
```const jsDate = new Date(`${date[1]} ${date[2]}`);
const month = jsDate.getMonth() + 1;```

Reply:
Thanks Github for making that a oneliner

Reply:
Heck you could even replace ${date[1]} with 1 and it would work